.github/workflows/ci.yml

```diff
@@ -32,38 +28,66 @@ env:
   FORCE_COLOR: 1
 
 jobs:
+  identify-jobs-to-run:
+    name: 'Analyze changes'
+    if: ${{ !cancelled() && github.event_name == 'pull_request' }}
+    runs-on: 'ubuntu-20.04'
+    outputs:
+      needs-code-validation: ${{ steps.target-changes.outputs.needs-code-validation }}
+      needs-changelog-validation: ${{ steps.target-changes.outputs.needs-changelog-validation }}
+      needs-markdown-validation:  ${{ steps.target-changes.outputs.needs-markdown-validation }}
+    steps:
+      - uses: 'dorny/paths-filter@de90cc6fb38fc0963ad72b210f1f284cd68cea36'
```

Comment:
I wonder if it would make more sense to have this as a step in the `project-jobs` job instead. Run this step first, then the 'Build Matrix/Spwan...' based on the output of this one.

Reply:
It also orchestrates linting markdown and verifying changelogs jobs.

I also like your idea: if we move the step inside building jobs, we can represent linting markdown and verifying changelogs as spawned by `project-jobs`. To get things rolling, I'll make a follow-up PR to implement this specific change.

Reply:
Done in 89663eb5d9726c65f26530cfa4977f2edcc3fe6b